L1Trigger/L1TTrackMatch/plugins/DisplacedVertexProducer.cc

```diff
@@ -0,0 +1,221 @@
+      selectedTPs.push_back(my_tp);
+    }
+  }
+  //std::cout<<"num selected tracks: "<<selectedTracks.size()<<std::endl;
+  sort(selectedTracks.begin(), selectedTracks.end(), ComparePtTrack); 
+  std::unique_ptr<l1t::DisplacedTrackVertexCollection> product(new std::vector<l1t::DisplacedTrackVertex>());
+  for(int i=0; i<int(selectedTracks.size()-1); i++){
+    for(int j=i+1; j<int(selectedTracks.size()); j++){
+      if(dist_TPs(selectedTracks[i],selectedTracks[j])!=0) continue;
+      Double_t x_dv_trk = -9999.0;
+      Double_t y_dv_trk = -9999.0;
+      Double_t z_dv_trk = -9999.0;
+      edm::Ptr<TrackingParticle> tp_i = selectedTPs[i];
+      edm::Ptr<TrackingParticle> tp_j = selectedTPs[j];
+      bool isReal = false;
+      if(!tp_i.isNull() && !tp_j.isNull()){
+	bool isHard_i = false;
+	bool isHard_j = false;
+	if(tp_i->genParticles().size() && tp_j->genParticles().size()){
+	  isHard_i = tp_i->genParticles()[0]->isHardProcess() || tp_i->genParticles()[0]->fromHardProcessFinalState();
+	  isHard_j = tp_j->genParticles()[0]->isHardProcess() || tp_j->genParticles()[0]->fromHardProcessFinalState();
+	}
+	
+	if(tp_i->eventId().event()==0 && tp_j->eventId().event()==0 && fabs(tp_i->vx()-tp_j->vx())<0.0001 && fabs(tp_i->vy()-tp_j->vy())<0.0001 && fabs(tp_i->vz()-tp_j->vz())<0.0001 && isHard_i && isHard_j){
+	  isReal = true;
+	}
+      }
+      
+      int inTraj = calcVertex(selectedTracks[i],selectedTracks[j],x_dv_trk,y_dv_trk,z_dv_trk);
+      Vertex_Parameters vertex = Vertex_Parameters(x_dv_trk, y_dv_trk, z_dv_trk, selectedTracks[i], selectedTracks[j]);
+      l1t::DisplacedTrackVertex outputVertex = l1t::DisplacedTrackVertex(selectedTracks[i].index, selectedTracks[j].index, i, j, inTraj, vertex.d_T, vertex.R_T, vertex.cos_T, vertex.delta_z, vertex.x_dv, vertex.y_dv, vertex.z_dv, vertex.openingAngle, vertex.p_mag, fabs(i-j), isReal);
+      std::vector<std::string> ortinput_names;
+      std::vector<std::string> ortoutput_names;
+      cms::Ort::FloatArrays ortinput;
+      cms::Ort::FloatArrays ortoutputs;
+      float minD0 = vertex.a.d0;
+      if(fabs(vertex.b.d0)<fabs(minD0)) minD0 = vertex.b.d0;
+      //std::vector<float> Transformed_features = {vertex.delta_z, vertex.R_T, vertex.cos_T, vertex.d_T, vertex.chi2rzdofSum, float(vertex.numStubsSum), vertex.chi2rphidofSum, minD0, vertex.a.pt+vertex.b.pt};
+      std::vector<float> Transformed_features = {selectedTracks[i].pt, selectedTracks[j].pt, selectedTracks[i].eta, selectedTracks[j].eta, selectedTracks[i].phi, selectedTracks[j].phi, selectedTracks[i].d0, selectedTracks[j].d0, selectedTracks[i].z0, selectedTracks[j].z0, selectedTracks[i].chi2rz, selectedTracks[j].chi2rz, selectedTracks[i].bendchi2, selectedTracks[j].bendchi2, selectedTracks[i].MVA1, selectedTracks[j].MVA1, selectedTracks[i].MVA2, selectedTracks[j].MVA2, vertex.d_T, vertex.R_T, vertex.cos_T, vertex.delta_z};
+      //cms::Ort::ONNXRuntime Runtime(this->ONNXmodel_);  //Setup ONNX runtime
```

Comment:
This comment doesn't need to be here, in fact we should remove the similar comment from the TrackQuality.cc as the Runtime is now initialised (as you do) in the constructor

Reply:
Removed in commit [63cdfcf](https://github.com/cms-l1t-offline/cmssw/pull/1257/commits/63cdfcf8c0dbba752b18dac750196655f6cf1b35)